plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkProvider.java

```diff
@@ -16,90 +16,58 @@
 
 package com.google.cloud.tools.eclipse.sdk;
 
+import com.google.common.annotations.VisibleForTesting;
 import com.google.cloud.tools.appengine.cloudsdk.CloudSdk;
-import com.google.cloud.tools.appengine.cloudsdk.PathResolver;
 import com.google.cloud.tools.eclipse.sdk.internal.PreferenceConstants;
 import com.google.cloud.tools.eclipse.sdk.internal.PreferenceInitializer;
 
 import org.eclipse.e4.core.contexts.ContextFunction;
 import org.eclipse.jface.preference.IPreferenceStore;
 
-import java.io.File;
 import java.nio.file.Path;
+import java.nio.file.Paths;
 
 /**
  * Utility to find the Google Cloud SDK either at locations configured by the user or in standard
  * locations on the system.
  */
 public class CloudSdkProvider extends ContextFunction {
 
-  private final IPreferenceStore preferences;
-
-   public CloudSdkProvider(IPreferenceStore preferences) {
-     // todo drop the null check and use the no args constructor instead
-     if (preferences == null) {
-       preferences = PreferenceInitializer.getPreferenceStore();
-     }
-     this.preferences = preferences;
-   }
-   
-   public CloudSdkProvider() {
-     this(PreferenceInitializer.getPreferenceStore());
-   }
-
-  /**
-   * Return the {@link CloudSdk} instance from the configured or discovered Cloud SDK.
-   * 
-   * @return the configured {@link CloudSdk} or {@code null} if no SDK could be found
-   */
-  public CloudSdk getCloudSdk() {
-    return createBuilder(null).build();
+  private IPreferenceStore preferences = PreferenceInitializer.getPreferenceStore();
+  private Path location;
+  
+  public CloudSdkProvider() {}
+  
+  @VisibleForTesting
+  public CloudSdkProvider(IPreferenceStore preferences) {
```

Comment:
It turns out that this is tested from another package (src/com/google/cloud/tools/eclipse/sdk/internal), so it needs to be public.